{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a168c137",
   "metadata": {},
   "source": [
    "# Chapter 8: RNA-seq analysis\n",
    "\n",
    "```{epigraph}\n",
    "*\"Do the difficult things while they are easy and do the great things while they are small. A journey of a thousand miles must begin with a single step.\"*\n",
    "\n",
    "-- Lao Tzu\n",
    "```\n",
    "\n",
    "```{seealso}\n",
    "- [RNA-Seq](https://en.wikipedia.org/wiki/RNA-Seq)\n",
    "```\n",
    "\n",
    "In general, RNA-SEQ analysis involves three steps to process and analyze short read sequences produced by RNA-seq. In brief, the process is to map and summarize short read sequences, then normalize between samples and perform a statistical test of DE. Further biological insight can be gained by looking for patterns of expression changes within sets of genes and integrating the RNA-seq data with data from other sources.\n",
    "\n",
    "Several comparisons of RNA-seq and microarray data have now been made.\n",
    "The results are unanimous: sequencing has higher sensitivity and dynamic\n",
    "range, coupled with lower technical variation. Furthermore, comparisons\n",
    "have highlighted strong concordance between microarrays and sequencing\n",
    "in measures of both absolute and differential expression.\n",
    "\n",
    "Sequencing gives unprecedented detail about novel transcribed regions,\n",
    "allele-specific expression, RNA editing and alternative splicing.\n",
    "\n",
    "## Mapping \n",
    "\n",
    "To use RNA-seq data to compare expression between samples, it is\n",
    "necessary to turn millions of short reads into a quantification of\n",
    "expression. The first step in this procedure is the read mapping or\n",
    "alignment. At its simplest, the task of mapping is to find the unique\n",
    "location where a short read is identical to the reference. However, in\n",
    "reality the reference is never a perfect representation of the actual\n",
    "biological source of RNA being sequenced. In addition to sample-specific\n",
    "attributes such as SNPs and indels (insertions or deletions), there is\n",
    "also the consideration that the reads arise from a spliced transcriptome\n",
    "rather than a genome. Furthermore, short reads can sometimes align\n",
    "perfectly to multiple locations and can contain sequencing errors that\n",
    "have to be accounted for. Therefore, the real task is to find the\n",
    "location where each short read best matches the reference, while\n",
    "allowing for errors and structural variation.\n",
    "\n",
    "Although research into how best to align reads to a reference is\n",
    "ongoing, all solutions by necessity involve some compromise between the\n",
    "computational requirements of the algorithm and the fuzziness allowed in\n",
    "matching to the reference. Almost all short read aligners use a strategy\n",
    "of a first pass \\'heuristic\\' match, which quickly finds a reduced list\n",
    "of possible locations, followed by thorough evaluation of all candidate\n",
    "alignments by a complex \\'local alignment\\' algorithm. Without this\n",
    "initial heuristic search to reduce the number of potential alignment\n",
    "locations, performing local alignment of millions of short reads would\n",
    "be computationally impossible on current hardware.\n",
    "\n",
    "Current aligners enable fast heuristic matching by using either **hash\n",
    "tables** or the **Burrows Wheeler transform** (BWT). Hash-table aligners\n",
    "have the advantage of being easily extendable to detect complicated\n",
    "differences between read and reference, at the cost of ever increasing\n",
    "computational requirements. Alternatively, BWT-based aligners can map\n",
    "reads that closely match the reference very efficiently but are\n",
    "prohibitively slow once more complex misalignments are considered.\n",
    "\n",
    "## Summarizing mapping reads\n",
    "\n",
    "Having obtained genomic locations for as many reads as possible, the\n",
    "next task is to summarize and aggregate reads over some biologically\n",
    "meaningful unit, such as exons, transcripts or genes. The simplest and\n",
    "most common approach counts the number of reads overlapping the exons in\n",
    "a gene (for example,\n",
    "\\[[32](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3046478/#B32),[34](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3046478/#B34),[45](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3046478/#B45)\\]).\n",
    "However, a significant proportion of reads map to genomic regions\n",
    "outside annotated exons, even in well-annotated organisms, such as mouse\n",
    "and human. For example, Pickrell*et\n",
    "al. *\\[[35](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3046478/#B35)\\]\n",
    "found that about 15% of mapped reads were located outside annotated\n",
    "exons for their Nigerian HapMap samples and these extra-exonic reads\n",
    "were more likely to be cell-type-specific exons. Similarly,\n",
    "Figure 2a shows an example of transcription occurring outside annotated\n",
    "exons in the RNA-binding protein 39 (*RBM39*) gene in LNCaP prostate\n",
    "cancer cells. Reads from other normal tissue cell types are more limited\n",
    "to known exons, but also show evidence for transcription outside of\n",
    "known exons.\n",
    "\n",
    "## Normalization\n",
    "\n",
    "### Normalization methods differ for between- and within-library comparisons.\n",
    "\n",
    "Within-library normalization: Because longer transcripts have higher\n",
    "read counts (at the same expression level), a common method for\n",
    "within-library normalization is to divide the summarized counts by the\n",
    "length of the gene. The widely used RPKM (reads per kilobase of exon\n",
    "model per million mapped reads) accounts for both library size and gene\n",
    "length effects in within-sample comparisons.\n",
    "\n",
    "## Differential expression\n",
    "\n",
    "The goal of a DE analysis is to highlight genes that have changed significantly in abundance across experimental conditions. In general, this means taking a table of summarized count data for each library and performing statistical testing between samples of interest.\n",
    "\n",
    "In general, the Poisson distribution forms the basis for modeling\n",
    "RNA-seq count data. In an early RNA-seq study using a single source of\n",
    "RNA, goodness-of-fit statistics suggested that the distribution of\n",
    "counts across lanes for the majority of genes was indeed Poisson\n",
    "distributed. This has been independently confirmed using a technical\n",
    "experiment. However, biological variability is not captured well by the\n",
    "Poisson assumption. Hence, Poisson-based analyses for datasets with\n",
    "biological replicates will be prone to high false positive rates\n",
    "resulting from the underestimation of sampling error.\n",
    "\n",
    "To account for biological variability, the negative binomial\n",
    "distribution has been used, requiring an additional dispersion parameter\n",
    "to be estimated. A few variations of negative-binomial-based DE analysis\n",
    "of count data have emerged, including common dispersion models, sharing\n",
    "information over all genes using weighted likelihood, empirical\n",
    "estimation of the mean-variance relationship and an empirical Bayesian\n",
    "implementation using equivalence classes.\n",
    "\n",
    "Extensions to the Poisson model to include over-dispersion have also\n",
    "been proposed, through the generalized Poisson distribution or a\n",
    "two-stage Poisson model, which tests for differential expression in two\n",
    "modes depending on the evidence for over-dispersion in the data. Several\n",
    "tools for either simultaneous transcript discovery and quantification or\n",
    "alternative isoform expression analysis also perform DE analysis.\n",
    "\n",
    "## Systems biology: going beyond gene lists\n",
    "\n",
    "In many cases, creating lists of DE genes is not the final step of the\n",
    "analysis; further biological insight into an experimental system can be\n",
    "gained by looking at the expression changes of sets of genes. Many tools\n",
    "focusing on gene set testing, network inference and knowledge databases\n",
    "have been designed for analyzing lists of DE genes from microarray\n",
    "datasets.\n",
    "\n",
    "However, RNA-seq is affected by biases not present in microarray data.\n",
    "For example, gene length bias is an issue in RNA-seq data, in which\n",
    "longer genes have higher counts (at the same expression level). This\n",
    "results in greater statistical power to detect DE for long and highly\n",
    "expressed genes. These biases can dramatically affect the results of\n",
    "downstream analyses, such as testing Gene Ontology (GO) terms for\n",
    "enrichment among DE genes.\n",
    "\n",
    "## Detection of alternative splicing\n",
    "\n",
    "1.  **Aligners** capable of identifying splice sites from sequence reads\n",
    "    (aka splice aware aligners). These\n",
    "    include: [TopHat](http://tophat.cbcb.umd.edu/), [MapSplice](http://www.netlab.uky.edu/p/bioinfo/MapSplice), [SpliceMap](http://www.stanford.edu/group/wonglab/SpliceMap/), [HMMsplicer](http://derisilab.ucsf.edu/index.php?software=105), [GSNAP](http://research-pub.gene.com/gmap/), [STAR](https://code.google.com/p/rna-star/), [RUM](http://cbil.upenn.edu/RUM/), [SoapSplice](http://soap.genomics.org.cn/soapsplice.html),\n",
    "    etc. I saw a nice poster at AGBT13 that indicated that STAR performs\n",
    "    very well compared to several competitors.\n",
    "\n",
    "2.  **Transcriptome assemblers** that either perform de novo assembly of\n",
    "    transcripts from sequence reads or do so with the help of a\n",
    "    reference assembly (and perhaps even guided by known transcript\n",
    "    annotations). These\n",
    "    include: [Cufflinks](http://cufflinks.cbcb.umd.edu/), [Scripture](http://www.broadinstitute.org/software/scripture/), [Trinity](http://trinityrnaseq.sourceforge.net/), [Trans-ABySS](http://www.bcgsc.ca/platform/bioinfo/software/trans-abyss), [GRIT](http://grit-bio.org/),\n",
    "    etc. Of these, Cufflinks is probably the easiest to use while\n",
    "    Trinity and Trans ABySS seem to yield impressive results in the\n",
    "    hands of certain groups (particularly those that developed\n",
    "    them\\...).\n",
    "\n",
    "3.  **Alternative expression** tools that seek to identify isoform\n",
    "    expression differences between two or more conditions. These\n",
    "    include: [Cuffdiff](http://cufflinks.cbcb.umd.edu/manual.html#cuffdiff), [ALEXA-seq](http://www.alexaplatform.org/alexa_seq/), [MISO](http://genes.mit.edu/burgelab/miso/), [SplicingCompass](http://www.ichip.de/software/SplicingCompass.html), [Flux\n",
    "    Capacitor](http://flux.sammeth.net/capacitor.html), [JuncBASE](http://compbio.berkeley.edu/proj/juncbase/Home.html),[DEXSeq](http://bioconductor.org/packages/2.11/bioc/html/DEXSeq.html), [MATS](http://rnaseq-mats.sourceforge.net/), [SpliceR](http://www.bioconductor.org/packages/2.13/bioc/html/spliceR.html), [FineSplice](http://nar.oxfordjournals.org/content/early/2014/02/25/nar.gku166.full), [ARH-seq](http://nar.oxfordjournals.org/content/early/2014/06/11/nar.gku495.full),\n",
    "    etc.\n",
    "\n",
    "## Detection of gene fusion\n",
    "\n",
    "Fusion genes are identified by selecting all the gene-gene pairs for\n",
    "which there are at least two or three short read pairs such that one end\n",
    "aligns to one of the genes and the other to the other.\n",
    "\n",
    "To identify the exon-exon fusion junction, a database of artificial\n",
    "splice-site junctions was built by generating all the potential\n",
    "exon-exon combinations between gene A and gene B and B-A for each pair\n",
    "of candidate-fusion genes. Short reads that did not align on either the\n",
    "genome or the transcriptome were aligned against the junction database\n",
    "in order to locate the exact fusion point, that is, between which exons\n",
    "the gene fusion takes place. Junctions spanning short reads were\n",
    "required to align at least 10 bp to one exon.\n",
    "\n",
    "Here\\'s a list of fusion gene detection tools working with RNA-seq data\n",
    "\n",
    "-   [tophat-fusion](http://tophat.cbcb.umd.edu/fusion_index.html)\n",
    "\n",
    "-   [chimerascan](http://code.google.com/p/chimerascan/)\n",
    "\n",
    "-   [deFuse](http://sourceforge.net/apps/mediawiki/defuse/index.php?title=Main_Page)\n",
    "\n",
    "-   [FusionMap](http://www.omicsoft.com/fusionmap/)\n",
    "\n",
    "## RNA editing \n",
    "\n",
    "RNA editing is a molecular process through which some cells can make\n",
    "discrete changes to specific nucleotide sequences within a RNA molecule\n",
    "after it has been generated by RNA polymerase. Occasionally researches\n",
    "encounter a gene with a sequence of nucleotides that does not match\n",
    "exactly that in its RNA product: messenger RNA.\n",
    "\n",
    "[\\\n",
    "REDITOOLS: efficient RNA editing detection by RNA-SE](https://github.com/BioinfoUNIBA/REDItools)\n",
    "\n",
    "[RADAR: a rigorously annotated database of A-to-I RNA editing](https://pubmed.ncbi.nlm.nih.gov/24163250/)\n",
    "\n",
    "## Novel transcripts\n",
    "\n",
    "Alternatively, fusion genes or aberrations in cancer samples can be\n",
    "detected by finding novel transcripts in RNA-seq data."
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "formats": "md:myst",
   "text_representation": {
    "extension": ".md",
    "format_name": "myst",
    "format_version": 0.13,
    "jupytext_version": "1.11.5"
   }
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  },
  "source_map": [
   14
  ]
 },
 "nbformat": 4,
 "nbformat_minor": 5
}