
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Chapter 8: RNA-seq analysis &#8212; Statistical Inference in Bioinformatics</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/proof.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Chapter 9: Linear regression" href="chap9.html" />
    <link rel="prev" title="Chapter 7: Multiple tests" href="chap7.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="_static/logo.jpg" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Statistical Inference in Bioinformatics</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Table of Contents
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="preface.html">
   Preface
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Lecture
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="chap0.html">
   Chapter 0: Prerequisites
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap1.html">
   Chapter 1: Probability theory
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap2.html">
   Chapter 2: Discrete random variables
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap3.html">
   Chapter 3: Continuous random variables
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap4.html">
   Chapter 4: Multiple random variables
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap5.html">
   Chapter 5: Estimation theory
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap6.html">
   Chapter 6: Hypothesis testing
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap7.html">
   Chapter 7: Multiple tests
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Chapter 8: RNA-seq analysis
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap9.html">
   Chapter 9: Linear regression
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap10.html">
   Chapter 10: Monte Carlo simulation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap11.html">
   Chapter 11: Bootstrap
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap12.html">
   Chapter 12: Bayesian analysis
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap13.html">
   Chapter 13: Markov Chains
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap14.html">
   Chapter 14: Substitution models
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="chap15.html">
   Chapter 15: Phylogenetic models
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Lab
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="lab1.html">
   Lab 1: Introduction to R
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<div class="menu-dropdown menu-dropdown-launch-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Launch interactive content">
      <i class="fas fa-rocket"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://mybinder.org/v2/gh/executablebooks/jupyter-book/master?urlpath=tree/docs/chap8.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Binder"
>
  

<span class="headerbtn__icon-container">
  
    <img src="_static/images/logo_binder.svg">
  </span>
<span class="headerbtn__text-container">Binder</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/executablebooks/jupyter-book"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/executablebooks/jupyter-book/issues/new?title=Issue%20on%20page%20%2Fchap8.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="_sources/chap8.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download notebook file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-code"></i>
  </span>
<span class="headerbtn__text-container">.ipynb</span>
</a>

      </li>
      
      <li>
        <a href="_sources/chap8.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.md</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#mapping">
   Mapping
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#summarizing-mapping-reads">
   Summarizing mapping reads
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#normalization">
   Normalization
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#normalization-methods-differ-for-between-and-within-library-comparisons">
     Normalization methods differ for between- and within-library comparisons.
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#differential-expression">
   Differential expression
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#systems-biology-going-beyond-gene-lists">
   Systems biology: going beyond gene lists
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#detection-of-alternative-splicing">
   Detection of alternative splicing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#detection-of-gene-fusion">
   Detection of gene fusion
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#rna-editing">
   RNA editing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#novel-transcripts">
   Novel transcripts
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Chapter 8: RNA-seq analysis</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#mapping">
   Mapping
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#summarizing-mapping-reads">
   Summarizing mapping reads
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#normalization">
   Normalization
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#normalization-methods-differ-for-between-and-within-library-comparisons">
     Normalization methods differ for between- and within-library comparisons.
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#differential-expression">
   Differential expression
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#systems-biology-going-beyond-gene-lists">
   Systems biology: going beyond gene lists
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#detection-of-alternative-splicing">
   Detection of alternative splicing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#detection-of-gene-fusion">
   Detection of gene fusion
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#rna-editing">
   RNA editing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#novel-transcripts">
   Novel transcripts
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="chapter-8-rna-seq-analysis">
<h1>Chapter 8: RNA-seq analysis<a class="headerlink" href="#chapter-8-rna-seq-analysis" title="Permalink to this headline">#</a></h1>
<blockquote class="epigraph">
<div><p><em>“Do the difficult things while they are easy and do the great things while they are small. A journey of a thousand miles must begin with a single step.”</em></p>
<p class="attribution">—Lao Tzu</p>
</div></blockquote>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<ul class="simple">
<li><p><a class="reference external" href="https://en.wikipedia.org/wiki/RNA-Seq">RNA-Seq</a></p></li>
</ul>
</div>
<p>In general, RNA-SEQ analysis involves three steps to process and analyze short read sequences produced by RNA-seq. In brief, the process is to map and summarize short read sequences, then normalize between samples and perform a statistical test of DE. Further biological insight can be gained by looking for patterns of expression changes within sets of genes and integrating the RNA-seq data with data from other sources.</p>
<p>Several comparisons of RNA-seq and microarray data have now been made.
The results are unanimous: sequencing has higher sensitivity and dynamic
range, coupled with lower technical variation. Furthermore, comparisons
have highlighted strong concordance between microarrays and sequencing
in measures of both absolute and differential expression.</p>
<p>Sequencing gives unprecedented detail about novel transcribed regions,
allele-specific expression, RNA editing and alternative splicing.</p>
<section id="mapping">
<h2>Mapping<a class="headerlink" href="#mapping" title="Permalink to this headline">#</a></h2>
<p>To use RNA-seq data to compare expression between samples, it is
necessary to turn millions of short reads into a quantification of
expression. The first step in this procedure is the read mapping or
alignment. At its simplest, the task of mapping is to find the unique
location where a short read is identical to the reference. However, in
reality the reference is never a perfect representation of the actual
biological source of RNA being sequenced. In addition to sample-specific
attributes such as SNPs and indels (insertions or deletions), there is
also the consideration that the reads arise from a spliced transcriptome
rather than a genome. Furthermore, short reads can sometimes align
perfectly to multiple locations and can contain sequencing errors that
have to be accounted for. Therefore, the real task is to find the
location where each short read best matches the reference, while
allowing for errors and structural variation.</p>
<p>Although research into how best to align reads to a reference is
ongoing, all solutions by necessity involve some compromise between the
computational requirements of the algorithm and the fuzziness allowed in
matching to the reference. Almost all short read aligners use a strategy
of a first pass ‘heuristic’ match, which quickly finds a reduced list
of possible locations, followed by thorough evaluation of all candidate
alignments by a complex ‘local alignment’ algorithm. Without this
initial heuristic search to reduce the number of potential alignment
locations, performing local alignment of millions of short reads would
be computationally impossible on current hardware.</p>
<p>Current aligners enable fast heuristic matching by using either <strong>hash
tables</strong> or the <strong>Burrows Wheeler transform</strong> (BWT). Hash-table aligners
have the advantage of being easily extendable to detect complicated
differences between read and reference, at the cost of ever increasing
computational requirements. Alternatively, BWT-based aligners can map
reads that closely match the reference very efficiently but are
prohibitively slow once more complex misalignments are considered.</p>
</section>
<section id="summarizing-mapping-reads">
<h2>Summarizing mapping reads<a class="headerlink" href="#summarizing-mapping-reads" title="Permalink to this headline">#</a></h2>
<p>Having obtained genomic locations for as many reads as possible, the
next task is to summarize and aggregate reads over some biologically
meaningful unit, such as exons, transcripts or genes. The simplest and
most common approach counts the number of reads overlapping the exons in
a gene (for example,
[<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3046478/#B32">32</a>,<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3046478/#B34">34</a>,<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3046478/#B45">45</a>]).
However, a significant proportion of reads map to genomic regions
outside annotated exons, even in well-annotated organisms, such as mouse
and human. For example, Pickrell*et
al. *[<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3046478/#B35">35</a>]
found that about 15% of mapped reads were located outside annotated
exons for their Nigerian HapMap samples and these extra-exonic reads
were more likely to be cell-type-specific exons. Similarly,
Figure 2a shows an example of transcription occurring outside annotated
exons in the RNA-binding protein 39 (<em>RBM39</em>) gene in LNCaP prostate
cancer cells. Reads from other normal tissue cell types are more limited
to known exons, but also show evidence for transcription outside of
known exons.</p>
</section>
<section id="normalization">
<h2>Normalization<a class="headerlink" href="#normalization" title="Permalink to this headline">#</a></h2>
<section id="normalization-methods-differ-for-between-and-within-library-comparisons">
<h3>Normalization methods differ for between- and within-library comparisons.<a class="headerlink" href="#normalization-methods-differ-for-between-and-within-library-comparisons" title="Permalink to this headline">#</a></h3>
<p>Within-library normalization: Because longer transcripts have higher
read counts (at the same expression level), a common method for
within-library normalization is to divide the summarized counts by the
length of the gene. The widely used RPKM (reads per kilobase of exon
model per million mapped reads) accounts for both library size and gene
length effects in within-sample comparisons.</p>
</section>
</section>
<section id="differential-expression">
<h2>Differential expression<a class="headerlink" href="#differential-expression" title="Permalink to this headline">#</a></h2>
<p>The goal of a DE analysis is to highlight genes that have changed significantly in abundance across experimental conditions. In general, this means taking a table of summarized count data for each library and performing statistical testing between samples of interest.</p>
<p>In general, the Poisson distribution forms the basis for modeling
RNA-seq count data. In an early RNA-seq study using a single source of
RNA, goodness-of-fit statistics suggested that the distribution of
counts across lanes for the majority of genes was indeed Poisson
distributed. This has been independently confirmed using a technical
experiment. However, biological variability is not captured well by the
Poisson assumption. Hence, Poisson-based analyses for datasets with
biological replicates will be prone to high false positive rates
resulting from the underestimation of sampling error.</p>
<p>To account for biological variability, the negative binomial
distribution has been used, requiring an additional dispersion parameter
to be estimated. A few variations of negative-binomial-based DE analysis
of count data have emerged, including common dispersion models, sharing
information over all genes using weighted likelihood, empirical
estimation of the mean-variance relationship and an empirical Bayesian
implementation using equivalence classes.</p>
<p>Extensions to the Poisson model to include over-dispersion have also
been proposed, through the generalized Poisson distribution or a
two-stage Poisson model, which tests for differential expression in two
modes depending on the evidence for over-dispersion in the data. Several
tools for either simultaneous transcript discovery and quantification or
alternative isoform expression analysis also perform DE analysis.</p>
</section>
<section id="systems-biology-going-beyond-gene-lists">
<h2>Systems biology: going beyond gene lists<a class="headerlink" href="#systems-biology-going-beyond-gene-lists" title="Permalink to this headline">#</a></h2>
<p>In many cases, creating lists of DE genes is not the final step of the
analysis; further biological insight into an experimental system can be
gained by looking at the expression changes of sets of genes. Many tools
focusing on gene set testing, network inference and knowledge databases
have been designed for analyzing lists of DE genes from microarray
datasets.</p>
<p>However, RNA-seq is affected by biases not present in microarray data.
For example, gene length bias is an issue in RNA-seq data, in which
longer genes have higher counts (at the same expression level). This
results in greater statistical power to detect DE for long and highly
expressed genes. These biases can dramatically affect the results of
downstream analyses, such as testing Gene Ontology (GO) terms for
enrichment among DE genes.</p>
</section>
<section id="detection-of-alternative-splicing">
<h2>Detection of alternative splicing<a class="headerlink" href="#detection-of-alternative-splicing" title="Permalink to this headline">#</a></h2>
<ol class="simple">
<li><p><strong>Aligners</strong> capable of identifying splice sites from sequence reads
(aka splice aware aligners). These
include: <a class="reference external" href="http://tophat.cbcb.umd.edu/">TopHat</a>, <a class="reference external" href="http://www.netlab.uky.edu/p/bioinfo/MapSplice">MapSplice</a>, <a class="reference external" href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a>, <a class="reference external" href="http://derisilab.ucsf.edu/index.php?software=105">HMMsplicer</a>, <a class="reference external" href="http://research-pub.gene.com/gmap/">GSNAP</a>, <a class="reference external" href="https://code.google.com/p/rna-star/">STAR</a>, <a class="reference external" href="http://cbil.upenn.edu/RUM/">RUM</a>, <a class="reference external" href="http://soap.genomics.org.cn/soapsplice.html">SoapSplice</a>,
etc. I saw a nice poster at AGBT13 that indicated that STAR performs
very well compared to several competitors.</p></li>
<li><p><strong>Transcriptome assemblers</strong> that either perform de novo assembly of
transcripts from sequence reads or do so with the help of a
reference assembly (and perhaps even guided by known transcript
annotations). These
include: <a class="reference external" href="http://cufflinks.cbcb.umd.edu/">Cufflinks</a>, <a class="reference external" href="http://www.broadinstitute.org/software/scripture/">Scripture</a>, <a class="reference external" href="http://trinityrnaseq.sourceforge.net/">Trinity</a>, <a class="reference external" href="http://www.bcgsc.ca/platform/bioinfo/software/trans-abyss">Trans-ABySS</a>, <a class="reference external" href="http://grit-bio.org/">GRIT</a>,
etc. Of these, Cufflinks is probably the easiest to use while
Trinity and Trans ABySS seem to yield impressive results in the
hands of certain groups (particularly those that developed
them…).</p></li>
<li><p><strong>Alternative expression</strong> tools that seek to identify isoform
expression differences between two or more conditions. These
include: <a class="reference external" href="http://cufflinks.cbcb.umd.edu/manual.html#cuffdiff">Cuffdiff</a>, <a class="reference external" href="http://www.alexaplatform.org/alexa_seq/">ALEXA-seq</a>, <a class="reference external" href="http://genes.mit.edu/burgelab/miso/">MISO</a>, <a class="reference external" href="http://www.ichip.de/software/SplicingCompass.html">SplicingCompass</a>, <a class="reference external" href="http://flux.sammeth.net/capacitor.html">Flux
Capacitor</a>, <a class="reference external" href="http://compbio.berkeley.edu/proj/juncbase/Home.html">JuncBASE</a>,<a class="reference external" href="http://bioconductor.org/packages/2.11/bioc/html/DEXSeq.html">DEXSeq</a>, <a class="reference external" href="http://rnaseq-mats.sourceforge.net/">MATS</a>, <a class="reference external" href="http://www.bioconductor.org/packages/2.13/bioc/html/spliceR.html">SpliceR</a>, <a class="reference external" href="http://nar.oxfordjournals.org/content/early/2014/02/25/nar.gku166.full">FineSplice</a>, <a class="reference external" href="http://nar.oxfordjournals.org/content/early/2014/06/11/nar.gku495.full">ARH-seq</a>,
etc.</p></li>
</ol>
</section>
<section id="detection-of-gene-fusion">
<h2>Detection of gene fusion<a class="headerlink" href="#detection-of-gene-fusion" title="Permalink to this headline">#</a></h2>
<p>Fusion genes are identified by selecting all the gene-gene pairs for
which there are at least two or three short read pairs such that one end
aligns to one of the genes and the other to the other.</p>
<p>To identify the exon-exon fusion junction, a database of artificial
splice-site junctions was built by generating all the potential
exon-exon combinations between gene A and gene B and B-A for each pair
of candidate-fusion genes. Short reads that did not align on either the
genome or the transcriptome were aligned against the junction database
in order to locate the exact fusion point, that is, between which exons
the gene fusion takes place. Junctions spanning short reads were
required to align at least 10 bp to one exon.</p>
<p>Here’s a list of fusion gene detection tools working with RNA-seq data</p>
<ul class="simple">
<li><p><a class="reference external" href="http://tophat.cbcb.umd.edu/fusion_index.html">tophat-fusion</a></p></li>
<li><p><a class="reference external" href="http://code.google.com/p/chimerascan/">chimerascan</a></p></li>
<li><p><a class="reference external" href="http://sourceforge.net/apps/mediawiki/defuse/index.php?title=Main_Page">deFuse</a></p></li>
<li><p><a class="reference external" href="http://www.omicsoft.com/fusionmap/">FusionMap</a></p></li>
</ul>
</section>
<section id="rna-editing">
<h2>RNA editing<a class="headerlink" href="#rna-editing" title="Permalink to this headline">#</a></h2>
<p>RNA editing is a molecular process through which some cells can make
discrete changes to specific nucleotide sequences within a RNA molecule
after it has been generated by RNA polymerase. Occasionally researches
encounter a gene with a sequence of nucleotides that does not match
exactly that in its RNA product: messenger RNA.</p>
<p><a class="reference external" href="https://github.com/BioinfoUNIBA/REDItools"><br />
REDITOOLS: efficient RNA editing detection by RNA-SE</a></p>
<p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/24163250/">RADAR: a rigorously annotated database of A-to-I RNA editing</a></p>
</section>
<section id="novel-transcripts">
<h2>Novel transcripts<a class="headerlink" href="#novel-transcripts" title="Permalink to this headline">#</a></h2>
<p>Alternatively, fusion genes or aberrations in cancer samples can be
detected by finding novel transcripts in RNA-seq data.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="chap7.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Chapter 7: Multiple tests</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="chap9.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Chapter 9: Linear regression</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Liang Liu<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>