

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Chapter 9: Linear regression &#8212; Statistical Inference in Bioinformatics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/proof.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'chap9';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Chapter 10: Monte Carlo simulation" href="chap10.html" />
    <link rel="prev" title="Chapter 8: RNA-seq analysis" href="chap8.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="intro.html">
  
  
  
  
    
    
      
    
    
    <img src="_static/logo.jpg" class="logo__image only-light" alt="Logo image"/>
    <script>document.write(`<img src="_static/logo.jpg" class="logo__image only-dark" alt="Logo image"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Table of Contents      
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="preface.html">Preface</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Lecture</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="chap0.html">Chapter 0: Prerequisites</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap1.html">Chapter 1: Probability theory</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap2.html">Chapter 2: Discrete random variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap3.html">Chapter 3: Continuous random variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap4.html">Chapter 4: Multiple random variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap5.html">Chapter 5: Estimation theory</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap6.html">Chapter 6: Hypothesis testing</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap7.html">Chapter 7: Multiple tests</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap8.html">Chapter 8: RNA-seq analysis</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Chapter 9: Linear regression</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap10.html">Chapter 10: Monte Carlo simulation</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap11.html">Chapter 11: Bootstrap</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap12.html">Chapter 12: Bayesian analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap13.html">Chapter 13: Markov Chains</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap14.html">Chapter 14: Substitution models</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap15.html">Chapter 15: Phylogenetic models</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Lab</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="lab1.html">Lab 1: Introduction to R</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab2.html">Lab 2: Generating random numbers</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab3.html">Lab 3: The law of large numbers</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab4.html">Lab 4: BLAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab5.html">Lab 5: Optimization algorithms</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab6.html">Lab 6: Hypothesis testing</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab7.html">Lab 7: Multiple tests</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab8.html">Lab 8: RNA-seq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab9.html">Lab 9: Linear regression</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab10.html">Lab 10: Monte Carlo simulation</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab11.html">Lab 11: Bootstrap</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab12.html">Lab 12: Markov Chain Monte Carlo algorithm</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab13.html">Lab 13: Markov chains</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab14.html">Lab 14: Substitution models</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab15.html">Lab 15: Phylogenetic tree reconstruction</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/executablebooks/jupyter-book" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/executablebooks/jupyter-book/issues/new?title=Issue%20on%20page%20%2Fchap9.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/chap9.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Chapter 9: Linear regression</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#simple-linear-regression">Simple linear regression</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#least-square-estimates">Least square estimates</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#maximum-likelihood-estimates">Maximum likelihood estimates</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#testing-the-linear-relationship">Testing the linear relationship</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multiple-linear-regression">Multiple linear regression</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#least-square-estimates-of-beta">Least square estimates of <span class="math notranslate nohighlight">\(\beta\)</span></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Maximum likelihood estimates</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transformation">Transformation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#variable-selection">Variable selection</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#f-test">F-test</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#likelihood-ratio-test-lrt">Likelihood ratio test (LRT)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#akaike-information-criterion-aic">Akaike Information Criterion (AIC)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#bayes-information-criterion-bic">Bayes Information Criterion (BIC)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#algorithms-for-variable-selection">Algorithms for variable selection</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#forward-stepwise-selection">Forward stepwise selection</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#backward-stepwise-selection">Backward stepwise selection</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#forward-and-backward-stepwise-selection">Forward and backward stepwise selection</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#least-absolute-shrinkage-and-selection-operator-lasso">Least absolute shrinkage and selection operator (LASSO)</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#generalized-linear-regression">Generalized linear regression</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#logistic-model">Logistic model</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#loglinear-model-counts">Loglinear model (counts)</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="chapter-9-linear-regression">
<h1>Chapter 9: Linear regression<a class="headerlink" href="#chapter-9-linear-regression" title="Permalink to this heading">#</a></h1>
<blockquote class="epigraph">
<div><p><em>“Do the difficult things while they are easy and do the great things while they are small. A journey of a thousand miles must begin with a single step.”</em></p>
<p class="attribution">—Lao Tzu</p>
</div></blockquote>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<ul class="simple">
<li><p><a class="reference external" href="https://en.wikipedia.org/wiki/Linear_regression">Linear regression</a></p></li>
</ul>
</div>
<section id="simple-linear-regression">
<h2>Simple linear regression<a class="headerlink" href="#simple-linear-regression" title="Permalink to this heading">#</a></h2>
<div class="admonition-model-assumptions admonition">
<p class="admonition-title">Model assumptions</p>
<p>Simple linear regression is a linear model with a single explanatory variable <span class="math notranslate nohighlight">\(x\)</span>. The model assumes a linear relationship between the response variable <span class="math notranslate nohighlight">\(y\)</span> and the explanatory variable <span class="math notranslate nohighlight">\(x\)</span>, i.e.,</p>
<p><span class="math notranslate nohighlight">\(y_{i}=\beta_0+\beta_1 x_{i}+\epsilon_{i}\)</span></p>
<p><span class="math notranslate nohighlight">\(x\)</span> : the explanatary variable. <span class="math notranslate nohighlight">\(x_i's\)</span> are fixed.</p>
<p><span class="math notranslate nohighlight">\(y\)</span>: the response variable.</p>
<p><span class="math notranslate nohighlight">\(\epsilon\)</span> : the error term. The model assumes that <span class="math notranslate nohighlight">\(\epsilon_{i}{ }^{\prime} s\)</span> are iid random variables with the normal <span class="math notranslate nohighlight">\(\left(0, \sigma^{2}\right)\)</span> density.</p>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">data</span><span class="p">(</span><span class="n">trees</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">trees</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 3</caption>
<thead>
	<tr><th></th><th scope=col>Girth</th><th scope=col>Height</th><th scope=col>Volume</th></tr>
	<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>1</th><td> 8.3</td><td>70</td><td>10.3</td></tr>
	<tr><th scope=row>2</th><td> 8.6</td><td>65</td><td>10.3</td></tr>
	<tr><th scope=row>3</th><td> 8.8</td><td>63</td><td>10.2</td></tr>
	<tr><th scope=row>4</th><td>10.5</td><td>72</td><td>16.4</td></tr>
	<tr><th scope=row>5</th><td>10.7</td><td>81</td><td>18.8</td></tr>
	<tr><th scope=row>6</th><td>10.8</td><td>83</td><td>19.7</td></tr>
</tbody>
</table>
</div></div>
</div>
<p>The probability distribution of <span class="math notranslate nohighlight">\(y_{i}\)</span> is <span class="math notranslate nohighlight">\(\operatorname{Normal}\left(\beta_0+\beta_1 x_{i}, \sigma^{2}\right)\)</span>. The expectation of <span class="math notranslate nohighlight">\(y\)</span> is a linear function of <span class="math notranslate nohighlight">\(x\)</span>, i.e., <span class="math notranslate nohighlight">\(E(y)\)</span> and <span class="math notranslate nohighlight">\(x\)</span> have a linear relationship,</p>
<div class="math notranslate nohighlight" id="equation-1">
<span class="eqno">(1)<a class="headerlink" href="#equation-1" title="Permalink to this equation">#</a></span>\[E(y) = \beta_0+\beta_1x\]</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">plot</span><span class="p">(</span><span class="n">Volume</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Height</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">trees</span><span class="p">,</span><span class="w"> </span><span class="n">pch</span><span class="o">=</span><span class="m">16</span><span class="p">)</span>
<span class="nf">abline</span><span class="p">(</span><span class="nf">lm</span><span class="p">(</span><span class="n">Volume</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Height</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">trees</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&#39;red&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/d9c244afe040eb65f8795e892be36eb065daed77634eaefaa79410883e79fb9b.png" src="_images/d9c244afe040eb65f8795e892be36eb065daed77634eaefaa79410883e79fb9b.png" />
</div>
</div>
<p>The intercept <span class="math notranslate nohighlight">\(\beta_0\)</span> and slope <span class="math notranslate nohighlight">\(\beta_1\)</span> are two parameters to estimate from data. Given <span class="math notranslate nohighlight">\(\beta_0\)</span> and <span class="math notranslate nohighlight">\(\beta_1\)</span>, the equation (1) can be used to predict the value of <span class="math notranslate nohighlight">\(y\)</span> for a new value of <span class="math notranslate nohighlight">\(x\)</span>, i.e.,</p>
<div class="math notranslate nohighlight" id="equation-2">
<span class="eqno">(2)<a class="headerlink" href="#equation-2" title="Permalink to this equation">#</a></span>\[y_{new} = \beta_0+\beta_1x_{new}\]</div>
<section id="least-square-estimates">
<h3>Least square estimates<a class="headerlink" href="#least-square-estimates" title="Permalink to this heading">#</a></h3>
<p>We estimate <span class="math notranslate nohighlight">\(\beta_0\)</span> and <span class="math notranslate nohighlight">\(\beta_1\)</span> by minimizing the sum of squared errors with respect to <span class="math notranslate nohighlight">\(\beta_0\)</span> and <span class="math notranslate nohighlight">\(\beta_1\)</span>,</p>
<div class="math notranslate nohighlight">
\[
\sum_{i=1}^{n}\left(y_{i}-\left(\beta_0+\beta_1 x_{i}\right)\right)^{2}
\]</div>
<p>Taking the derivative of the sum of squared errors with respect to <span class="math notranslate nohighlight">\(\beta_0\)</span> and <span class="math notranslate nohighlight">\(\beta_1\)</span>, we have</p>
<div class="math notranslate nohighlight">
\[
\frac{\partial \sum_{i=1}^{n}\left(y_{i}-\left(\beta_0+\beta_1 x_{i}\right)\right)^{2}}{\partial \beta_0}=0
\]</div>
<p>and</p>
<div class="math notranslate nohighlight">
\[
\frac{\partial \sum_{i=1}^{n}\left(y_{i}-\left(\beta_0+\beta_1 x_{i}\right)\right)^{2}}{\partial \beta_1}=0
\]</div>
<p>Thus,</p>
<div class="math notranslate nohighlight">
\[\begin{split}
\begin{equation}
\begin{cases}
\hat{\beta_0}=\bar{y}-\beta_1 \bar{x}\\
\hat{\beta_1}=\frac{\sum\left(x_{i}-\bar{x}\right)\left(y_{i}-\bar{y}\right)}{\sum\left(x_{i}-\bar{x}\right)^{2}}\\
\widehat{\sigma^{2}}=\frac{\sum\left(y_{i}-\hat{y}\right)^{2}}{n-2}
\end{cases}
\end{equation}
\end{split}\]</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">lm</span><span class="p">(</span><span class="n">Volume</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Height</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">trees</span><span class="p">)</span>

<span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">trees</span><span class="o">$</span><span class="n">Height</span>
<span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">trees</span><span class="o">$</span><span class="n">Volume</span>

<span class="n">beta1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">((</span><span class="n">x</span><span class="o">-</span><span class="nf">mean</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="o">*</span><span class="p">(</span><span class="n">y</span><span class="o">-</span><span class="nf">mean</span><span class="p">(</span><span class="n">y</span><span class="p">)))</span><span class="o">/</span><span class="nf">sum</span><span class="p">((</span><span class="n">x</span><span class="o">-</span><span class="nf">mean</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="o">^</span><span class="m">2</span><span class="p">)</span>
<span class="n">beta0</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">mean</span><span class="p">(</span><span class="n">y</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">beta1</span><span class="o">*</span><span class="nf">mean</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>

<span class="n">beta0</span>
<span class="n">beta1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Call:
lm(formula = Volume ~ Height, data = trees)

Coefficients:
(Intercept)       Height  
    -87.124        1.543  
</pre></div>
</div>
<div class="output text_html">-87.1236135388527</div><div class="output text_html">1.54334975369458</div></div>
</div>
</section>
<section id="maximum-likelihood-estimates">
<h3>Maximum likelihood estimates<a class="headerlink" href="#maximum-likelihood-estimates" title="Permalink to this heading">#</a></h3>
<p>The first step is to determine the likelihood function</p>
<div class="math notranslate nohighlight">
\[
f\left(y_{1}, y_{2}, \ldots, y_{n}\right)=\prod_{i=1}^{n} \frac{1}{\sqrt{2 \pi \sigma^{2}}} e^{-\frac{\sum\left(y_{i}-\left(\beta_0+\beta_1 x_{i}\right)\right)^{2}}{2 \sigma^{2}}}
\]</div>
<p>Next, we find the log-likelihood function,</p>
<div class="math notranslate nohighlight">
\[
n \log \left(\frac{1}{\sqrt{2 \pi \sigma^{2}}}\right)-\frac{\sum\left(y_{i}-\left(\beta_0+\beta_1 x_{i}\right)\right)^{2}}{2 \sigma^{2}}
\]</div>
<p>The log-likelihood is maximized if <span class="math notranslate nohighlight">\(\sum_{i=1}^{n}\left(y_{i}-\left(\beta_0+\beta_1 x_{i}\right)\right)^{2}\)</span> is minimized. Thus, the maximum likelihood estimates are identical with the least square estimates of <span class="math notranslate nohighlight">\(\beta_0\)</span> and <span class="math notranslate nohighlight">\(\beta_1\)</span>.</p>
</section>
<section id="testing-the-linear-relationship">
<h3>Testing the linear relationship<a class="headerlink" href="#testing-the-linear-relationship" title="Permalink to this heading">#</a></h3>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>If <span class="math notranslate nohighlight">\(\beta_1=0\)</span>, we say that <span class="math notranslate nohighlight">\(Y\)</span> and <span class="math notranslate nohighlight">\(X\)</span> do not have a linear relationship</p>
<p>If <span class="math notranslate nohighlight">\(\beta_1&gt;0\)</span>, we say that <span class="math notranslate nohighlight">\(Y\)</span> and <span class="math notranslate nohighlight">\(X\)</span> are positively correlated</p>
<p>If <span class="math notranslate nohighlight">\(\beta_1&lt;0\)</span>, we say that <span class="math notranslate nohighlight">\(Y\)</span> and <span class="math notranslate nohighlight">\(X\)</span> are negatively correlated</p>
</div>
<p>The likelihood ratio test (LRT) can be used to test if <span class="math notranslate nohighlight">\(Y\)</span> and <span class="math notranslate nohighlight">\(X\)</span> have a linear relationship.</p>
<p><span class="math notranslate nohighlight">\(\mathrm{H}_{0}: \beta_1=0\)</span> and <span class="math notranslate nohighlight">\(\mathrm{H}_{1}: \beta_1 \neq 0\)</span></p>
<p>The test statistic is <span class="math notranslate nohighlight">\(t=2 \log \left(l_{1}\right)-2 \log \left(l_{0}\right)\)</span>, where <span class="math notranslate nohighlight">\(\log \left(l_{0}\right)\)</span> is the loglikelihood score of the null model</p>
<div class="math notranslate nohighlight">
\[\log \left(l_{0}\right)=n \log \left(\frac{1}{\sqrt{2 \pi \sigma^{2}}}\right)-\frac{\sum\left(y_{i}-\beta_0\right)^{2}}{2 \sigma^{2}}=n \log \left(\frac{1}{\sqrt{2 \pi \hat{\sigma}^{2}}}\right)-\frac{\sum\left(y_{i}-\hat{\beta_0}\right)^{2}}{2 \hat{\sigma}^{2}}\]</div>
<p><span class="math notranslate nohighlight">\(\log \left(l_{1}\right)\)</span> is the loglikelihood score of the alternative model, i.e.,</p>
<div class="math notranslate nohighlight">
\[\begin{split}
\begin{equation}
\begin{split}
\log \left(l_{1}\right) &amp;=n \log \left(\frac{1}{\sqrt{2 \pi \sigma^{2}}}\right)-\frac{\sum\left(y_{i}-\left(\beta_0+\beta_1 x_{i}\right)\right)^{2}}{2 \sigma^{2}} \\
&amp;=n \log \left(\frac{1}{\sqrt{2 \pi \hat{\sigma}^{2}}}\right)-\frac{\sum\left(y_{i}-\left(\hat{\beta_0}+\hat{\beta_1} x_{i}\right)\right)^{2}}{2 \hat{\sigma}^{2}}
\end{split}
\end{equation}
\end{split}\]</div>
<p><span class="math notranslate nohighlight">\(H_{0}\)</span> has one free parameter <span class="math notranslate nohighlight">\(\beta_0\)</span>, while <span class="math notranslate nohighlight">\(H_{1}\)</span> has two free parameters <span class="math notranslate nohighlight">\(\beta_0\)</span> and <span class="math notranslate nohighlight">\(\beta_1\)</span>. Thus, the null distribution of the test statistic <span class="math notranslate nohighlight">\(t\)</span> is the chi-square distribution with 1 degree of freedom.</p>
<p>Rejection region: we reject the null if <span class="math notranslate nohighlight">\(t&gt;a\)</span> where <span class="math notranslate nohighlight">\(a\)</span> is the 95% quantile of the <span class="math notranslate nohighlight">\(\chi^2\)</span> distribution with 1 degree of freedom.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">result</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">Volume</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Height</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">trees</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Call:
lm(formula = Volume ~ Height, data = trees)

Residuals:
    Min      1Q  Median      3Q     Max 
-21.274  -9.894  -2.894  12.068  29.852 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -87.1236    29.2731  -2.976 0.005835 ** 
Height        1.5433     0.3839   4.021 0.000378 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 13.4 on 29 degrees of freedom
Multiple R-squared:  0.3579,	Adjusted R-squared:  0.3358 
F-statistic: 16.16 on 1 and 29 DF,  p-value: 0.0003784
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="multiple-linear-regression">
<h2>Multiple linear regression<a class="headerlink" href="#multiple-linear-regression" title="Permalink to this heading">#</a></h2>
<div class="admonition-model-assumptions admonition">
<p class="admonition-title">Model assumptions</p>
<p><span class="math notranslate nohighlight">\(y_{i}=\beta_{0}+\beta_{1} x_{1 i}+\beta_{2} x_{2 i}+\cdots+\beta_{p} x_{p i}+\varepsilon_{i}\)</span></p>
<p><span class="math notranslate nohighlight">\(x_{i}\)</span> : the explanatary variables. We assume <span class="math notranslate nohighlight">\(x_i's\)</span> are fixed.</p>
<p><span class="math notranslate nohighlight">\(y\)</span> : the response variable, which is a random variable.</p>
<p><span class="math notranslate nohighlight">\(\epsilon\)</span> : the error term. The model assumes that <span class="math notranslate nohighlight">\(\epsilon_{i}^{\prime} s\)</span> are independent and have the same probability distribution normal <span class="math notranslate nohighlight">\(\left(0, \sigma^{2}\right)\)</span>.</p>
</div>
<p>The probability distribution of <span class="math notranslate nohighlight">\(y_i\)</span> is normal with mean <span class="math notranslate nohighlight">\(\left(\beta_{0}+\beta_{1} X_{1 i}+\beta_{2} X_{2 i}+\cdots+\right.\)</span> <span class="math notranslate nohighlight">\(\left.\beta_{p} X_{p i}\right)\)</span> and variance <span class="math notranslate nohighlight">\(\sigma^{2}\)</span>, in which <span class="math notranslate nohighlight">\(\left(\beta_{0}, \beta_{1}, \ldots, \beta_{p}\right)\)</span> and <span class="math notranslate nohighlight">\(\sigma^{2}\)</span> are parameters to estimate. The matrix representation of the multiple linear regression with <span class="math notranslate nohighlight">\(p\)</span> predictors <span class="math notranslate nohighlight">\(\left(X_{1}, \ldots, X_{p}\right)\)</span> is given by</p>
<div class="math notranslate nohighlight">
\[
Y=X \beta+\epsilon
\]</div>
<p>where <span class="math notranslate nohighlight">\(Y=(y_1,\dots,y_n)\)</span> is a vector of size <span class="math notranslate nohighlight">\(n\)</span>, <span class="math notranslate nohighlight">\(X\)</span> is a <span class="math notranslate nohighlight">\(n \times(p+1)\)</span> matrix, and <span class="math notranslate nohighlight">\(\beta=\left(\beta_{0}, \beta_{1}, \ldots, \beta_{p}\right)\)</span> is a vector of size <span class="math notranslate nohighlight">\((p+1)\)</span>.</p>
<section id="least-square-estimates-of-beta">
<h3>Least square estimates of <span class="math notranslate nohighlight">\(\beta\)</span><a class="headerlink" href="#least-square-estimates-of-beta" title="Permalink to this heading">#</a></h3>
<p>We estimate <span class="math notranslate nohighlight">\(\beta=\left(\beta_{0}, \beta_{1}, \ldots, \beta_{p}\right)\)</span> by minimizing the sum of squared errors</p>
<div class="math notranslate nohighlight">
\[
SSE=\sum_{i=1}^{n}\left(y_{i}-\left(\beta_{0}+\beta_{1} x_{1 i}+\beta_{2} x_{2 i}+\cdots+\beta_{p} x_{p i}\right)\right)^{2}
\]</div>
<p>The matrix representation of the sum of squared error is <span class="math notranslate nohighlight">\(SSE=(Y-X\beta)'(Y-X\beta)\)</span>, which is equal to <span class="math notranslate nohighlight">\(Y'Y-Y'X\beta-\beta'X'Y+\beta'X'X\beta\)</span>. To minimize the sum of squared error, we take the first derivative with respect to <span class="math notranslate nohighlight">\(\beta\)</span> and set it to be 0, i.e., <span class="math notranslate nohighlight">\(-2X'Y+2X'X\beta=0\)</span>, indicating that <span class="math notranslate nohighlight">\(\beta = (X'X)^{-1}X'Y\)</span>. Thus, the least square estimate of <span class="math notranslate nohighlight">\(\beta\)</span> is <span class="math notranslate nohighlight">\(\hat{\beta}=(X'X)^{-1}X'Y\)</span> where <span class="math notranslate nohighlight">\(X'\)</span> is the transpose of the matrix <span class="math notranslate nohighlight">\(X\)</span>.</p>
</section>
<section id="id1">
<h3>Maximum likelihood estimates<a class="headerlink" href="#id1" title="Permalink to this heading">#</a></h3>
<p>The likelihood function is given by</p>
<div class="math notranslate nohighlight">
\[
f\left(y_{1}, y_{2}, \ldots, y_{n}\right)=\prod_{i=1}^{n} \frac{1}{\sqrt{2 \pi \sigma^{2}}} e^{-\frac{\left(y_{i}-\left(\beta_{0}+\beta_{1} x_{1 i}+\beta_{2} x_{2 i}+\cdots+\beta_{p} x_{p i}\right)\right)^{2}}{2 \sigma^{2}}}
\]</div>
<p>Since maximizing the likelihood function is equivalent to minimizing <span class="math notranslate nohighlight">\(\left(y_{i}-\left(\beta_{0}+\right.\right.\)</span> <span class="math notranslate nohighlight">\(\left.\left.\beta_{1} X_{1 i}+\beta_{2} X_{2 i}+\cdots+\beta_{p} X_{p i}\right)\right)^{2}\)</span>, the MLEs of <span class="math notranslate nohighlight">\(\beta\)</span> are identical with the least square estimates.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">result</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">Volume</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Girth</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">Height</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">trees</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Call:
lm(formula = Volume ~ Girth + Height, data = trees)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.4065 -2.6493 -0.2876  2.2003  8.4847 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -57.9877     8.6382  -6.713 2.75e-07 ***
Girth         4.7082     0.2643  17.816  &lt; 2e-16 ***
Height        0.3393     0.1302   2.607   0.0145 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 3.882 on 28 degrees of freedom
Multiple R-squared:  0.948,	Adjusted R-squared:  0.9442 
F-statistic:   255 on 2 and 28 DF,  p-value: &lt; 2.2e-16
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="transformation">
<h2>Transformation<a class="headerlink" href="#transformation" title="Permalink to this heading">#</a></h2>
<p>The response variable <span class="math notranslate nohighlight">\(Y\)</span> may not have a linear relationship with <span class="math notranslate nohighlight">\(X\)</span>. For example, if the true relation is <span class="math notranslate nohighlight">\(E(Y)=\beta_0+\beta_1 X^{2}\)</span>, we can transform <span class="math notranslate nohighlight">\(Z=X^{2}\)</span> and fit a linear model for <span class="math notranslate nohighlight">\(Y\)</span> and <span class="math notranslate nohighlight">\(Z\)</span> (or <span class="math notranslate nohighlight">\(X^{2}\)</span>)</p>
<p>In general, if <span class="math notranslate nohighlight">\(E(Y)=g(X)\)</span>, we can fit a linear regression for <span class="math notranslate nohighlight">\(Y\)</span> and a function <span class="math notranslate nohighlight">\(g(X)\)</span> of <span class="math notranslate nohighlight">\(X\)</span>, where the function <span class="math notranslate nohighlight">\(g(X)\)</span> is determined by</p>
<ol class="arabic simple">
<li><p>The residual plots to discover the relationship between <span class="math notranslate nohighlight">\(Y\)</span> and <span class="math notranslate nohighlight">\(X\)</span> and find <span class="math notranslate nohighlight">\(g(X)\)</span></p></li>
<li><p>Log-transformation for <span class="math notranslate nohighlight">\(X\)</span> and <span class="math notranslate nohighlight">\(Y\)</span></p></li>
<li><p>Box-Cox transformation: <span class="math notranslate nohighlight">\(Y=\left\{\begin{array}{l}\frac{Y^{\lambda}-1}{\lambda} \text {, if } \lambda \neq 0 \\ \log (Y) \text {, if } \lambda=0\end{array}\right.\)</span> and find the optimal <span class="math notranslate nohighlight">\(\lambda\)</span> using R.</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">MASS</span><span class="p">)</span>

<span class="nf">boxcox </span><span class="p">(</span><span class="n">Volume</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">log</span><span class="p">(</span><span class="n">Height</span><span class="p">)</span><span class="o">+</span><span class="nf">log</span><span class="p">(</span><span class="n">Girth</span><span class="p">),</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">trees</span><span class="p">,</span><span class="w"> </span><span class="n">lambda</span><span class="w"> </span><span class="o">=</span><span class="nf">seq</span><span class="p">(</span><span class="m">-0.25</span><span class="p">,</span><span class="m">0.25</span><span class="p">,</span><span class="w"> </span><span class="n">length</span><span class="o">=</span><span class="m">10</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/272235cd5825dda3f71a883f84bc98304afb818ab9aeb47a7dda6db248552b0f.png" src="_images/272235cd5825dda3f71a883f84bc98304afb818ab9aeb47a7dda6db248552b0f.png" />
</div>
</div>
</section>
<section id="variable-selection">
<h2>Variable selection<a class="headerlink" href="#variable-selection" title="Permalink to this heading">#</a></h2>
<p>Variable selection is intended to select the “best” subset of predictors. We want to explain the data in the simplest way. The principle is that among several plausible explanations for a phenomenon, the simplest is best. Applied to regression analysis, this implies that the smallest model that fits the data is best.</p>
<p>Considering the linear regression model with <span class="math notranslate nohighlight">\(p\)</span> predictors <span class="math notranslate nohighlight">\(\left\{X_{1}, \ldots, X_{p}\right\}\)</span></p>
<div class="math notranslate nohighlight">
\[
Y=\beta_{0}+\beta_{1} X_{1}+\cdots+\beta_{p} X_{p}+\epsilon
\]</div>
<p>We want to select the smallest subset of predictors <span class="math notranslate nohighlight">\(\left\{X_{1}, \ldots, X_{p}\right\}\)</span> that can fit the data, i.e., we are interested in testing</p>
<div class="math notranslate nohighlight">
\[
H_{0}: \beta_{j}=0 \text { vs } H_{1}: \beta_{j} \neq 0
\]</div>
<p>With the normality assumption, we may use t-test to test whether <span class="math notranslate nohighlight">\(\beta_{j}=0\)</span>. In general, to select a subset of predictors, we need to test if a subset <span class="math notranslate nohighlight">\(\beta_{s}\)</span> of <span class="math notranslate nohighlight">\(\beta\)</span> are 0 , i.e.,</p>
<div class="math notranslate nohighlight">
\[
H_{0}: \beta_{s}=0 \text { vs } H_{1}: \beta_{s} \neq 0
\]</div>
<p>The model under <span class="math notranslate nohighlight">\(H_{0}\)</span> is called reduced model with <span class="math notranslate nohighlight">\(p_{s}+1\)</span> coefficients <span class="math notranslate nohighlight">\(\beta\)</span> and the model under <span class="math notranslate nohighlight">\(H_{1}\)</span> is called full model with <span class="math notranslate nohighlight">\((p+1)\)</span> coefficients <span class="math notranslate nohighlight">\(\beta\)</span>. Note that <span class="math notranslate nohighlight">\(p_{s}&lt;p\)</span>, and the null model is nested in the alternative model. The likelihood ratio test can determine if a subset <span class="math notranslate nohighlight">\(\beta_{s}\)</span> are 0.</p>
<section id="f-test">
<h3>F-test<a class="headerlink" href="#f-test" title="Permalink to this heading">#</a></h3>
<p>The residual sum-of-squares <span class="math notranslate nohighlight">\(\operatorname{RSS}(\beta)\)</span> is defined as:</p>
<div class="math notranslate nohighlight">
\[
RSS(\beta)=\sum_{i=1}^{n}\left(y_{i}-\widehat{y}_{l}\right)^{2}=\sum_{i=1}^{n}\left(y_{i}-x_{i} \beta\right)^{2}
\]</div>
<p>Let <span class="math notranslate nohighlight">\(\mathrm{RSS}_{1}\)</span> correspond to the full model with <span class="math notranslate nohighlight">\(p+1\)</span> parameters, and <span class="math notranslate nohighlight">\(\mathrm{RSS}_{0}\)</span> correspond to the nested model with <span class="math notranslate nohighlight">\(p_{0}+1\)</span> parameters. The <span class="math notranslate nohighlight">\(F\)</span> statistic measures the reduction of RSS per additional parameter in the full model,</p>
<div class="math notranslate nohighlight">
\[
F=\frac{\left(R S S_{0}-R S S_{1}\right) /\left(p_{1}-p_{0}\right)}{R S S_{1} /\left(n-p_{1}-1\right)}
\]</div>
<p>Under the normality assumption, the null distribution of <span class="math notranslate nohighlight">\(F\)</span> test statistic is the <span class="math notranslate nohighlight">\(F\)</span> distribution with degrees of freedom <span class="math notranslate nohighlight">\(\left(p_{1}-p_{0}\right)\)</span> and <span class="math notranslate nohighlight">\(\left(n-p_{1}-1\right)\)</span>. Thus, we reject <span class="math notranslate nohighlight">\(H_{0}\)</span> if <span class="math notranslate nohighlight">\(F&gt;\)</span> a, where <span class="math notranslate nohighlight">\(a\)</span> is the <span class="math notranslate nohighlight">\((1-\alpha)\)</span> quantile of the <span class="math notranslate nohighlight">\(F\)</span> distribution with degrees of freedom <span class="math notranslate nohighlight">\(\left(p_{1}-p_{0}\right)\)</span> and <span class="math notranslate nohighlight">\(\left(n-p_{1}-1\right)\)</span>.</p>
</section>
<section id="likelihood-ratio-test-lrt">
<h3>Likelihood ratio test (LRT)<a class="headerlink" href="#likelihood-ratio-test-lrt" title="Permalink to this heading">#</a></h3>
<p>Let <span class="math notranslate nohighlight">\(L_{1}\)</span> be the maximum value of the likelihood of the full model. Let <span class="math notranslate nohighlight">\(L_{0}\)</span> be the maximum value of the likelihood of the nested model. The likelihood ratio test statistic is</p>
<div class="math notranslate nohighlight">
\[
t=2 \log \left(L_{1}\right)-2 \log \left(L_{0}\right)
\]</div>
<p>The null distribution of test statistic <span class="math notranslate nohighlight">\(t\)</span> is asymptotically <span class="math notranslate nohighlight">\(\chi^{2}\)</span> distribution with <span class="math notranslate nohighlight">\(\left(p_{1}-p_{0}\right)\)</span> degrees of freedom where <span class="math notranslate nohighlight">\(p_1\)</span> is the number of parameters in the alternative model and <span class="math notranslate nohighlight">\(p_0\)</span> is the number of parameters in the null model. Thus, we reject <span class="math notranslate nohighlight">\(\mathrm{H}_{0}\)</span> if <span class="math notranslate nohighlight">\(t&gt;\mathrm{a}\)</span>, where <span class="math notranslate nohighlight">\(a\)</span> is the <span class="math notranslate nohighlight">\((1-\alpha)\)</span> quantile of the <span class="math notranslate nohighlight">\(\chi^{2}\)</span> distribution with <span class="math notranslate nohighlight">\(\left(p_{1}-p_{0}\right)\)</span> degrees of freedom.</p>
</section>
<section id="akaike-information-criterion-aic">
<h3>Akaike Information Criterion (AIC)<a class="headerlink" href="#akaike-information-criterion-aic" title="Permalink to this heading">#</a></h3>
<p>The LRT tends to favor the complex model, because LRT is solely based on the likelihood score and the complex model always has a higher likelihood score. AIC is a more general measure of “model fit” by penalizing the complexity of the model to avoid overfitting the data,</p>
<div class="math notranslate nohighlight">
\[
\text { AIC }=-2 \text { loglikelihood }+2 p
\]</div>
<p>where <span class="math notranslate nohighlight">\(p\)</span> is the number of parameters, measuring the complexity of the model, and loglikelihood measures goodness of fit of the model to the data. Given a collection of putative models, the best model is the one with the lowest <span class="math notranslate nohighlight">\(AIC\)</span>.</p>
</section>
<section id="bayes-information-criterion-bic">
<h3>Bayes Information Criterion (BIC)<a class="headerlink" href="#bayes-information-criterion-bic" title="Permalink to this heading">#</a></h3>
<p>AIC tends to overfit models when the sample size is small. Another information criterion which penalizes complex models more severely is</p>
<div class="math notranslate nohighlight">
\[
\text { BIC }=-2 \text { loglikelihood }+p * \log (n)
\]</div>
<p>Given a collection of putative models, the best model is the one with the lowest BIC.</p>
</section>
<section id="algorithms-for-variable-selection">
<h3>Algorithms for variable selection<a class="headerlink" href="#algorithms-for-variable-selection" title="Permalink to this heading">#</a></h3>
<p>An exhaustive search for the subset may not be feasible if <span class="math notranslate nohighlight">\(p\)</span> is very large. There are two main alternatives:</p>
<section id="forward-stepwise-selection">
<h4>Forward stepwise selection<a class="headerlink" href="#forward-stepwise-selection" title="Permalink to this heading">#</a></h4>
<p>The algorithm begins with a naive model <span class="math notranslate nohighlight">\(y = \beta_0\)</span> that does not include any explanatary variables <span class="math notranslate nohighlight">\(x\)</span>. Then, we add one explanatary variable at a time. We always choose from the rest of the variables the one that yields the best accuracy in prediction when added to the pool of already selected variables. This accuracy can be measured by the F statistic, LRT, AIC, BIC, etc. The algorithm stops when no predictors can be added.</p>
</section>
<section id="backward-stepwise-selection">
<h4>Backward stepwise selection<a class="headerlink" href="#backward-stepwise-selection" title="Permalink to this heading">#</a></h4>
<p>The algorithm begins with the full model including all predictors and then removes one predictor at a time.</p>
</section>
<section id="forward-and-backward-stepwise-selection">
<h4>Forward and backward stepwise selection<a class="headerlink" href="#forward-and-backward-stepwise-selection" title="Permalink to this heading">#</a></h4>
<p>Mixing forward and backward selection to find the optimal subset of predictors.</p>
</section>
<section id="least-absolute-shrinkage-and-selection-operator-lasso">
<h4>Least absolute shrinkage and selection operator (LASSO)<a class="headerlink" href="#least-absolute-shrinkage-and-selection-operator-lasso" title="Permalink to this heading">#</a></h4>
<p>The algorithm minimizes the residual sum-of-square <span class="math notranslate nohighlight">\(\operatorname{RSS}(\beta)=\sum_{i=1}^{n}\left(y_{i}-\widehat{y}_{l}\right)^{2}\)</span>, subject to <span class="math notranslate nohighlight">\(\sum_{i=1}^{p}\left|\beta_{i}\right| \leq c\)</span>, in which <span class="math notranslate nohighlight">\(c\)</span> is a pre-specified parameter. This procedure can automatically shrink some <span class="math notranslate nohighlight">\(\beta\)</span> to 0 .</p>
</section>
</section>
</section>
<section id="generalized-linear-regression">
<h2>Generalized linear regression<a class="headerlink" href="#generalized-linear-regression" title="Permalink to this heading">#</a></h2>
<p>In linear regression model, the dependent variable <span class="math notranslate nohighlight">\(Y\)</span> is assumed to have a normal distribution with mean <span class="math notranslate nohighlight">\(X \beta\)</span>. In generalized linear regression, the normality assumption is relaxed to allow <span class="math notranslate nohighlight">\(Y\)</span> to have other probability distributions (Binomial, Poisson, etc).</p>
<section id="logistic-model">
<h3>Logistic model<a class="headerlink" href="#logistic-model" title="Permalink to this heading">#</a></h3>
<p>In the logistic model, the response variable <span class="math notranslate nohighlight">\(Y_{i}=(0,1)\)</span> is a Bernoulli random variable with probability <span class="math notranslate nohighlight">\(p_{i}\)</span> satisfying <span class="math notranslate nohighlight">\(\log \left(\frac{p_{i}}{1-p_{i}}\right)=X_{i} \beta\)</span>, which is called the link function. It indicates that <span class="math notranslate nohighlight">\(p_{i}=\frac{e^{X_{i} \beta}}{1+e^{X_{i} \beta}}\)</span>. The logistic model can be used to predict probabilities given the covariates <span class="math notranslate nohighlight">\(X\)</span>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">&quot;https://stats.idre.ucla.edu/stat/data/binary.csv&quot;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 4</caption>
<thead>
	<tr><th></th><th scope=col>admit</th><th scope=col>gre</th><th scope=col>gpa</th><th scope=col>rank</th></tr>
	<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>1</th><td>0</td><td>380</td><td>3.61</td><td>3</td></tr>
	<tr><th scope=row>2</th><td>1</td><td>660</td><td>3.67</td><td>3</td></tr>
	<tr><th scope=row>3</th><td>1</td><td>800</td><td>4.00</td><td>1</td></tr>
	<tr><th scope=row>4</th><td>1</td><td>640</td><td>3.19</td><td>4</td></tr>
	<tr><th scope=row>5</th><td>0</td><td>520</td><td>2.93</td><td>4</td></tr>
	<tr><th scope=row>6</th><td>1</td><td>760</td><td>3.00</td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<p>Since <span class="math notranslate nohighlight">\(Y_{i}\)</span> ‘s are independent, the likelihood function is given by</p>
<div class="math notranslate nohighlight">
\[
\prod_{i=1}^{n} p_{i}^{Y_{i}}\left(1-p_{i}\right)^{1-Y_{i}}=\prod_{i=1}^{n}\left(\frac{e^{X_{i} \beta}}{1+e^{X_{i} \beta}}\right)^{Y_{i}}\left(\frac{1}{1+e^{X_{i} \beta}}\right)^{1-Y_{i}}
\]</div>
<p>The MLE of <span class="math notranslate nohighlight">\(\beta\)</span> is given by maximizing this likelihood function with respect to <span class="math notranslate nohighlight">\(\beta\)</span>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">admit</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">gre</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">gpa</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">rank</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;binomial&quot;</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Call:
glm(formula = admit ~ gre + gpa + rank, family = &quot;binomial&quot;, 
    data = data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.5802  -0.8848  -0.6382   1.1575   2.1732  

Coefficients:
             Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept) -3.449548   1.132846  -3.045  0.00233 ** 
gre          0.002294   0.001092   2.101  0.03564 *  
gpa          0.777014   0.327484   2.373  0.01766 *  
rank        -0.560031   0.127137  -4.405 1.06e-05 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 499.98  on 399  degrees of freedom
Residual deviance: 459.44  on 396  degrees of freedom
AIC: 467.44

Number of Fisher Scoring iterations: 4
</pre></div>
</div>
</div>
</div>
<p>We can predict the admission probability for a new observation <span class="math notranslate nohighlight">\(X\)</span> using the equation <span class="math notranslate nohighlight">\(p=\frac{e^{X \beta}}{1+e^{X \beta}}\)</span></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">newdata1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gre</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">590</span><span class="p">,</span><span class="w"> </span><span class="n">gpa</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3.9</span><span class="p">,</span><span class="w"> </span><span class="n">rank</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
<span class="nf">predict</span><span class="p">(</span><span class="n">result</span><span class="p">,</span><span class="w"> </span><span class="n">newdata</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">newdata1</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;response&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><strong>1:</strong> 0.592475691285015</div></div>
</div>
</section>
<section id="loglinear-model-counts">
<h3>Loglinear model (counts)<a class="headerlink" href="#loglinear-model-counts" title="Permalink to this heading">#</a></h3>
<p><span class="math notranslate nohighlight">\(Y_{i}\)</span> is a Poisson random variable with <span class="math notranslate nohighlight">\(\log \left(E\left(Y_{i}\right)\right)=X_{i} \beta\)</span>. It indicates that <span class="math notranslate nohighlight">\(Y_{\mathrm{i}}\)</span> is a Poisson random variable with mean <span class="math notranslate nohighlight">\(\lambda_{i}=e^{X_{i} \beta}\)</span>. Thus, the likelihood function is given by</p>
<div class="math notranslate nohighlight">
\[
\prod_{i=1}^{n} \frac{e^{-\lambda_{i}} \lambda_{i}^{Y_{i}}}{Y_{i} !}=\prod_{i=1}^{n} \frac{e^{-e^{X_{i} \beta}}\left(e^{X_{i} \beta}\right)^{Y_{i}}}{Y_{i} !}
\]</div>
<p>The MLE of <span class="math notranslate nohighlight">\(\beta\)</span> are obtained by maximizing the likelihood function.</p>
<p>We may use LRT to test <span class="math notranslate nohighlight">\(\mathrm{H}_{0}: \beta_{1}=0\)</span> vs <span class="math notranslate nohighlight">\(\mathrm{H}_{1}: \beta_{1} \neq 0\)</span> for the logistic and loglinear models. The procedures of variable selection can also apply to the logistic and loglinear models.</p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="chap8.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Chapter 8: RNA-seq analysis</p>
      </div>
    </a>
    <a class="right-next"
       href="chap10.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Chapter 10: Monte Carlo simulation</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#simple-linear-regression">Simple linear regression</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#least-square-estimates">Least square estimates</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#maximum-likelihood-estimates">Maximum likelihood estimates</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#testing-the-linear-relationship">Testing the linear relationship</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multiple-linear-regression">Multiple linear regression</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#least-square-estimates-of-beta">Least square estimates of <span class="math notranslate nohighlight">\(\beta\)</span></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Maximum likelihood estimates</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transformation">Transformation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#variable-selection">Variable selection</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#f-test">F-test</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#likelihood-ratio-test-lrt">Likelihood ratio test (LRT)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#akaike-information-criterion-aic">Akaike Information Criterion (AIC)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#bayes-information-criterion-bic">Bayes Information Criterion (BIC)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#algorithms-for-variable-selection">Algorithms for variable selection</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#forward-stepwise-selection">Forward stepwise selection</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#backward-stepwise-selection">Backward stepwise selection</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#forward-and-backward-stepwise-selection">Forward and backward stepwise selection</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#least-absolute-shrinkage-and-selection-operator-lasso">Least absolute shrinkage and selection operator (LASSO)</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#generalized-linear-regression">Generalized linear regression</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#logistic-model">Logistic model</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#loglinear-model-counts">Loglinear model (counts)</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Liang Liu
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2022.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>