

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Chapter 8: RNA-seq analysis &#8212; Statistical Inference in Bioinformatics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/proof.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'chap8';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Chapter 9: Linear regression" href="chap9.html" />
    <link rel="prev" title="Chapter 7: Multiple tests" href="chap7.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="intro.html">
  
  
  
  
    
    
      
    
    
    <img src="_static/logo.jpg" class="logo__image only-light" alt="Logo image"/>
    <script>document.write(`<img src="_static/logo.jpg" class="logo__image only-dark" alt="Logo image"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Table of Contents                              
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="preface.html">Preface</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Lecture</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="chap0.html">Chapter 0: Prerequisites</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap1.html">Chapter 1: Probability theory</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap2.html">Chapter 2: Discrete random variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap3.html">Chapter 3: Continuous random variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap4.html">Chapter 4: Multiple random variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap5.html">Chapter 5: Estimation theory</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap6.html">Chapter 6: Hypothesis testing</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap7.html">Chapter 7: Multiple tests</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Chapter 8: RNA-seq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap9.html">Chapter 9: Linear regression</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap10.html">Chapter 10: Monte Carlo simulation</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap11.html">Chapter 11: Bootstrap</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap12.html">Chapter 12: Bayesian analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap13.html">Chapter 13: Markov Chains</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap14.html">Chapter 14: Substitution models</a></li>
<li class="toctree-l1"><a class="reference internal" href="chap15.html">Chapter 15: Phylogenetic models</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Lab</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="lab1.html">Lab 1: Introduction to R</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab2.html">Lab 2: Generating random numbers</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab3.html">Lab 3: The law of large numbers</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab4.html">Lab 4: BLAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab5.html">Lab 5: Optimization algorithms</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab6.html">Lab 6: Hypothesis testing</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab7.html">Lab 7: Multiple tests</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab8.html">Lab 8: RNA-seq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab9.html">Lab 9: Linear regression</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab10.html">Lab 10: Monte Carlo simulation</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab11.html">Lab 11: Bootstrap</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab12.html">Lab 12: Markov Chain Monte Carlo algorithm</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab13.html">Lab 13: Markov chains</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab14.html">Lab 14: Substitution models</a></li>
<li class="toctree-l1"><a class="reference internal" href="lab15.html">Lab 15: Phylogenetic tree reconstruction</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/executablebooks/jupyter-book" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/executablebooks/jupyter-book/issues/new?title=Issue%20on%20page%20%2Fchap8.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/chap8.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Chapter 8: RNA-seq analysis</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mapping">Mapping</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#normalization">Normalization</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#differential-expression">Differential expression</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#detection-of-alternative-splicing">Detection of alternative splicing</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#detection-of-gene-fusion">Detection of gene fusion</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#rna-editing">RNA editing</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="chapter-8-rna-seq-analysis">
<h1>Chapter 8: RNA-seq analysis<a class="headerlink" href="#chapter-8-rna-seq-analysis" title="Permalink to this heading">#</a></h1>
<blockquote class="epigraph">
<div><p><em>“Do the difficult things while they are easy and do the great things while they are small. A journey of a thousand miles must begin with a single step.”</em></p>
<p class="attribution">—Lao Tzu</p>
</div></blockquote>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<ul class="simple">
<li><p><a class="reference external" href="https://en.wikipedia.org/wiki/RNA-Seq">RNA-Seq</a></p></li>
</ul>
</div>
<div class="mxgraph" style="max-width:100%;border:1px solid transparent;" data-mxgraph="{&quot;highlight&quot;:&quot;#0000ff&quot;,&quot;nav&quot;:true,&quot;resize&quot;:true,&quot;toolbar&quot;:&quot;zoom layers tags lightbox&quot;,&quot;edit&quot;:&quot;_blank&quot;,&quot;xml&quot;:&quot;&lt;mxfile host=\&quot;app.diagrams.net\&quot; modified=\&quot;2023-01-01T04:58:18.908Z\&quot; agent=\&quot;5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36\&quot; etag=\&quot;cEu5igzNmqGU--JUJIgs\&quot; version=\&quot;20.7.4\&quot; type=\&quot;dropbox\&quot;&gt;&lt;diagram id=\&quot;C5RBs43oDa-KdzZeNtuy\&quot; name=\&quot;Page-1\&quot;&gt;7Vrbdps6EP0ar5U+NIuLufixjp30nDZZOU3atI8yyKBGICLk2PTrjwQCgyEEp8ZO3L4k0ljoMnv2jGZgoJ8FqwsKIv+SuBAPNMVdDfTJQNPUoT7k/4QkySTW0MgEHkWuHLQW3KBfUAoVKV0gF8aVgYwQzFBUFTokDKHDKjJAKVlWh80Jrq4aAQ/WBDcOwHXpHXKZn0ltzVrLP0Lk+fnKqjnKfglAPlieJPaBS5YlkT4d6GeUEJa1gtUZxEJ5uV7u/knu8Od78+Lf/+IH8HX86fbq2/tssvNtHimOQGHIdju1lk39CPBC6kuelSW5AilZhC4UkygDfeyzAPOmyps/IWOJBBwsGOEiQplPPBIC/JmQSI6bk5DJYarow9D9IIDl/Rkmzn0mOkcYyzV4T463eS9mlNwX2IkJCiDEYAxmEI+Bc++lGz0jmFD+U0hCKKZyuTHIs6w3N11Lxx11KzGIyYI6sGWcLk0cUA+2zWdm48T+SnYqkbuAJICMJnwAhRgw9Fg1ZiA54RXj1rjzhoR+CzPQa2YQ+1xd6frAjVtsQuCx9BGDNxFIFbPkbqRqJ2X8+RHHHgZxLNF7Btw5N4oc0YGmz20HOk7xWOmXmW0MDWVrOB8hZXDVCkD+qyL9YJL7CelAlmunouaewi85lKHSE2bDGmY/YBtSR8lejjJNvovJTo2Rlgt+pAKFQyQFk1WxOdFLyr1rSBFHBFIp3LFDMOsO4ZJcOdGHs8cpTr7djen86/xCl877eYcg7S83tc7+Qc50TVDISkPIfB7zfVUCRz4mN/08Iia5RVvVZbKzyac2rLrY+csN3agZ+hWp2fnzpvKmmdC/UTaOs7YzSrU3o9yVLZk1WwpAFKHQqztOnwSzRfx8eKsGLG4j5yBAWGjkI8SPkCEHNARBgJEX8o7DARWup8k8+JJiZ/rEXPduU3Pkzr8WHKHqGtBqCo4j09KB2WNwtKoOQrMbYuOwITYaO4iNjc40D85t99ptPUajT2jwHSXTaGTPDmlsdaSxqh3yclnHYhEK1nHN/r1f5hSqcWik7e9+2cwh+4/gkKptc0E7FIfyXVYDV1pL4X9+LkKHIRLyJkYzCuSCr4BRPCTZ7rCJUbY2080+g5JmVTO2odYQlfS9MupVEKqSMunllOk9z5n0QrJ9zrQTkrYlR89yVB39Jkm7ZUCG2lsG1Hb8kuHIsKlwJ5CG0dQNUDiHXOFck7/H/hcGxRo1uxrAkxQebSSeXWsuZl8MrqcPB2HwCrHvpXZW7jBkb81b0clp2zdD1VFHihr7Yehog6HZAXpjqFan6Bu/LD0NXgeQrT3dldo2WYLC4Xplwl2eOCQSGwkXwQzS+N2xOErtqQBV9pTmPj1lbv9HToiGDLxZHft6vdO2yxIWIaEBwOgXyNIGE/PNj2eUtzzRWiLmI5FOxCCIMEyp8+X606XYXCjuGw8LEDKEYRuDlK0Z1Dlb2D2DDN2oMkgZnRrdOGQZPcGmvo7rRu9Xh863+4OSSK2H+Mk0JQTASYw4R/i8Cr9hxHGaiZ9Mphcx9N4VnLmCXrZNTZmhkAQIiDVPhGK/ZA/PQHIDH7L2ZBrDh6Oh19DWTqsE04x6iNJMo04vu7d0vP4G9RD0KvLxolO6z7ck4r3TMv9a51la7qu63Lx6/br3F8WXoKgfFMV6bcxF87ScwbifxGIncBVRGMdpvcODYevHDm/q9m4YVdeoNrjG/dY5GsrNEWD+EiSVgHccsUnXNtSvHFz99e+xhMGLxBWyJaH3fwIKTS+R94vCqIZCvAgCQIu06Tg0v1FlNbq+J+nt7b1WN/9DhPS+Q7PWOTTv69Vj6zafrh4oM+6VIKzUC0QKc3t5mTUWUQRpVjWgompwLMwxtM2sRrcbywZ2Q15jWafbFw54d/3BeVa8Xn+2r0//Bw==&lt;/diagram&gt;&lt;/mxfile&gt;&quot;}"></div>
<script type="text/javascript" src="https://viewer.diagrams.net/js/viewer-static.min.js"></script>
<p>RNA-seq involves sequencing RNA molecules to determine their structure and identify which genes are being expressed in a cell or tissue. Numerous computational approaches have been developed to process and analyze short read sequences produced by RNA-seq techniques. Short reads are mapped to the reference genome and quantified into copy numbers (exon, coding region, etc.). After normalization, the counts (copy numbers) are used to identify signicant genes by the differential expression (DE) analysis. Further biological insight is gained by pathway and gene network analyses. In comparison with microarray, RNA-seq has higher sensitivity and lower technical variation. More importantly, RNA-seq can discover novel transcribed regions, allele-specific expression, RNA editing and alternative splicing.</p>
<section id="mapping">
<h2>Mapping<a class="headerlink" href="#mapping" title="Permalink to this heading">#</a></h2>
<p>One common task in RNA-seq analysis is mapping the reads produced by the sequencing machine back to the reference genome or transcriptome to determine which genes are being transcribed and to quantify their expression levels. There are several tools available for mapping RNA-seq reads to a reference genome or transcriptome, including:</p>
<ul class="simple">
<li><p>HISAT2</p></li>
<li><p>TopHat</p></li>
<li><p>STAR</p></li>
<li><p>Subread</p></li>
<li><p>Bowtie</p></li>
</ul>
<p>These tools align the reads to the reference using various algorithms and heuristics, such as indexing the reference and using dynamic programming to find the best alignment for each read. The output of the mapping step is a set of alignments in a standardized format, such as BAM or SAM, which can then be processed further to identify differentially expressed genes, alternative splicing events, and other features of interest.</p>
<p>The process of read mapping, also known as alignment, involves identifying the unique location in a reference where a short read sequence matches. However, in practice, the reference may not be an exact representation of the RNA being sequenced due to sample-specific variations such as single nucleotide polymorphisms (SNPs) and insertions or deletions (indels). Additionally, reads may come from spliced transcripts rather than a genome, and may contain sequencing errors or align perfectly to multiple locations. Therefore, the goal of read mapping is to find the best match for each short read in the reference, taking into account these potential errors and variations.</p>
<p>Most short read aligners use a two-step approach: first, they use a fast heuristic algorithm to identify a list of potential alignment locations, and then they use a more computationally intensive local alignment algorithm to evaluate all of the candidate alignments. Fast heuristic matching is implemented in current aligners which either utilize hash tables or the Burrows Wheeler transform (BWT). Hash-table aligners are able to detect intricate discrepancies between the read and reference. On the other hand, BWT-based aligners are able to efficiently map reads that closely match the reference, but struggle to handle more complex misalignments and may become slow in these cases.</p>
</section>
<section id="normalization">
<h2>Normalization<a class="headerlink" href="#normalization" title="Permalink to this heading">#</a></h2>
<p>Longer transcripts have higher read counts. To correct for this, between and within sample normalization is often performed to normalize the read counts for each gene within (or between) a sample to the total number of reads obtained from that sample. This can be done using methods such as <strong>upper quartile normalization</strong>, or <strong>TMM</strong> (trimmed mean of M-values) normalization, or <strong>RPKM</strong> (reads per kilobase of exon model per million mapped reads), a widely used method for within-sample normalization.</p>
</section>
<section id="differential-expression">
<h2>Differential expression<a class="headerlink" href="#differential-expression" title="Permalink to this heading">#</a></h2>
<p>differential expression analysis is a method used to identify changes in gene expression between two or more groups of samples. It involves quantifying the abundance of RNA molecules in each sample, and then comparing the results to determine which genes are differentially expressed between the groups. There are several methods for performing differential expression analysis, including <strong>edgeR</strong>, <strong>DESeq2</strong>, and <strong>limma</strong>. These methods typically involve testing for statistical significance using tools such as the t-test or ANOVA, and correcting for multiple testing using techniques such as Bonferroni correction or false discovery rate (FDR) control. It is important to carefully design the experiment and choose appropriate statistical methods to ensure accurate results.</p>
<div class="admonition-a-list-of-methods-for-de-analysis admonition">
<p class="admonition-title">A list of methods for DE analysis</p>
<ul class="simple">
<li><p>DEGseg, assuming Poisson with a dispersion parameter</p></li>
<li><p>edgeR, baySeq, DEseq, assuming negative binomial with additional dispersion parameters</p></li>
</ul>
</div>
</section>
<section id="detection-of-alternative-splicing">
<h2>Detection of alternative splicing<a class="headerlink" href="#detection-of-alternative-splicing" title="Permalink to this heading">#</a></h2>
<ol class="arabic simple">
<li><p>Aligners identifying splice sites from sequence reads
(aka splice aware aligners): <a class="reference external" href="http://tophat.cbcb.umd.edu/">TopHat</a>, <a class="reference external" href="http://www.netlab.uky.edu/p/bioinfo/MapSplice">MapSplice</a>, <a class="reference external" href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a>, <a class="reference external" href="http://derisilab.ucsf.edu/index.php?software=105">HMMsplicer</a>, <a class="reference external" href="http://research-pub.gene.com/gmap/">GSNAP</a>, <a class="reference external" href="https://code.google.com/p/rna-star/">STAR</a>, <a class="reference external" href="http://cbil.upenn.edu/RUM/">RUM</a>, <a class="reference external" href="http://soap.genomics.org.cn/soapsplice.html">SoapSplice</a>,
etc.</p></li>
<li><p>Transcriptome assemblers: <a class="reference external" href="http://cufflinks.cbcb.umd.edu/">Cufflinks</a>, <a class="reference external" href="http://www.broadinstitute.org/software/scripture/">Scripture</a>, <a class="reference external" href="http://trinityrnaseq.sourceforge.net/">Trinity</a>, <a class="reference external" href="http://www.bcgsc.ca/platform/bioinfo/software/trans-abyss">Trans-ABySS</a>, <a class="reference external" href="http://grit-bio.org/">GRIT</a>.</p></li>
<li><p>Alternative expression tools identify isoform
expression: <a class="reference external" href="http://cufflinks.cbcb.umd.edu/manual.html#cuffdiff">Cuffdiff</a>, <a class="reference external" href="http://www.alexaplatform.org/alexa_seq/">ALEXA-seq</a>, <a class="reference external" href="http://genes.mit.edu/burgelab/miso/">MISO</a>, <a class="reference external" href="http://www.ichip.de/software/SplicingCompass.html">SplicingCompass</a>, <a class="reference external" href="http://flux.sammeth.net/capacitor.html">Flux
Capacitor</a>, <a class="reference external" href="http://compbio.berkeley.edu/proj/juncbase/Home.html">JuncBASE</a>,<a class="reference external" href="http://bioconductor.org/packages/2.11/bioc/html/DEXSeq.html">DEXSeq</a>, <a class="reference external" href="http://rnaseq-mats.sourceforge.net/">MATS</a>, <a class="reference external" href="http://www.bioconductor.org/packages/2.13/bioc/html/spliceR.html">SpliceR</a>, <a class="reference external" href="http://nar.oxfordjournals.org/content/early/2014/02/25/nar.gku166.full">FineSplice</a>, <a class="reference external" href="http://nar.oxfordjournals.org/content/early/2014/06/11/nar.gku495.full">ARH-seq</a>,
etc.</p></li>
</ol>
</section>
<section id="detection-of-gene-fusion">
<h2>Detection of gene fusion<a class="headerlink" href="#detection-of-gene-fusion" title="Permalink to this heading">#</a></h2>
<p>Fusion genes are detected by looking for pairs of genes that have at least two or three short read pairs where one end aligns with one gene and the other end aligns with the other gene. To identify the specific point where the genes fuse (the exon-exon junction), a database of potential exon-exon combinations between the two genes is created. Short reads that do not align to either the genome or transcriptome are then aligned against this junction database to determine the exact location of the fusion point. The alignment must span at least 10 base pairs with one exon in order to be considered a valid fusion gene.
A list of fusion gene detection tools</p>
<div class="admonition-a-list-of-methods-for-gene-fusion-detection admonition">
<p class="admonition-title">A list of methods for gene fusion detection</p>
<ul class="simple">
<li><p><a class="reference external" href="http://tophat.cbcb.umd.edu/fusion_index.html">tophat-fusion</a></p></li>
<li><p><a class="reference external" href="http://code.google.com/p/chimerascan/">chimerascan</a></p></li>
<li><p><a class="reference external" href="http://sourceforge.net/apps/mediawiki/defuse/index.php?title=Main_Page">deFuse</a></p></li>
<li><p><a class="reference external" href="http://www.omicsoft.com/fusionmap/">FusionMap</a></p></li>
</ul>
</div>
</section>
<section id="rna-editing">
<h2>RNA editing<a class="headerlink" href="#rna-editing" title="Permalink to this heading">#</a></h2>
<p>RNA editing is a process in which certain cells can modify specific nucleotide sequences within an RNA molecule after it has been synthesized by RNA polymerase. This can sometimes result in a gene having a sequence of nucleotides that does not exactly match the sequence of the corresponding messenger RNA.</p>
<div class="admonition-a-list-of-methods-for-rna-editing admonition">
<p class="admonition-title">A list of methods for RNA editing</p>
<p><a class="reference external" href="https://github.com/BioinfoUNIBA/REDItools">REDITOOLS: efficient RNA editing detection by RNA-SE</a>
<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/24163250/">RADAR: a rigorously annotated database of A-to-I RNA editing</a></p>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="chap7.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Chapter 7: Multiple tests</p>
      </div>
    </a>
    <a class="right-next"
       href="chap9.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Chapter 9: Linear regression</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mapping">Mapping</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#normalization">Normalization</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#differential-expression">Differential expression</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#detection-of-alternative-splicing">Detection of alternative splicing</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#detection-of-gene-fusion">Detection of gene fusion</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#rna-editing">RNA editing</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Liang Liu
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2022.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>